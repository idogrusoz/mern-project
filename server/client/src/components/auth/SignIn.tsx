import { Button, Grid, TextField } from "@material-ui/core";
import React, { useContext } from "react";
import { AuthContext } from "./AuthContext";

const SignIn = () => {
    const { email, setEmail, password, setPassword, setShowSignin, signin } = useContext(AuthContext);
    return (
        <>
            <Grid
                container
                justify="center"
                spacing={2}
                alignItems="center"
                direction="column"
                style={{ height: "100vh" }}
            >
                <Grid item>
                    <TextField
                        id="standard-basic"
                        label="Email"
                        type="Email"
                        onChange={(e: { target: { value: any } }) => setEmail(e.target.value)}
                        value={email}
                    />
                </Grid>
                <Grid item>
                    <TextField
                        id="standard-basic"
                        label="Password"
                        type="Password"
                        onChange={(e: { target: { value: any } }) => setPassword(e.target.value)}
                        value={password}
                    />
                </Grid>
                <Grid item>
                    <Button type="submit" variant="contained" color="primary" size="large" onClick={() => signin()}>
                        Sign In
                    </Button>
                </Grid>
                <Grid item>
                    <Button
                        type="submit"
                        variant="contained"
                        color="primary"
                        size="large"
                        onClick={() => {
                            console.log(1);
                            setShowSignin(false);
                        }}
                    >
                        Register
                    </Button>
                </Grid>
            </Grid>
        </>
    );
};

export default SignIn;
