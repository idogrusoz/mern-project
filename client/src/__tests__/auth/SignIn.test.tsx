import { Button, TextField } from "@material-ui/core";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import React from "react";
import SignIn, { SignInProps } from "../../components/Auth/SignIn";

describe("SignIn component", () => {
    const props: SignInProps = {
        setShowSignin: jest.fn((foo) => foo),
        signin: jest.fn(() => {}),
    };
    const wrapper = shallow(<SignIn {...props} />);
    it("renders without crash", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it("changes showsignin value", () => {
        wrapper.find(Button).at(1).simulate("click");
        expect(props.setShowSignin).toBeCalledWith(false);
    });
});
