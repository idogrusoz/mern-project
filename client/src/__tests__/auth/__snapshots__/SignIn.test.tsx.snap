// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component renders without crash 1`] = `
<form
  onSubmit={[Function]}
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    direction="column"
    justify="center"
    spacing={2}
    style={
      Object {
        "minHeight": " 100vh",
        "overflowX": "auto",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(TextField))
        error={false}
        id="standard-basic"
        label="Email"
        name="email"
        onChange={[Function]}
        type="Email"
        value=""
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(TextField))
        error={false}
        id="standard-basic"
        label="Password"
        name="password"
        onChange={[Function]}
        type="Password"
        value=""
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        size="large"
        type="submit"
        variant="contained"
      >
        Sign In
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
        size="large"
        variant="contained"
      >
        Register
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</form>
`;
